plumbing/bootstrap.py

```diff
@@ -74,88 +66,94 @@ def clean_error_message(message):
+    description_file = os.path.join(source_folder, DESCRIPTION_FILE)
+    datapackage_file = os.path.join(source_folder, DATAPACKAGE_FILE)
 
     with open(description_file) as stream:
         try:
             package = yaml.load(stream.read())
             package['name'] = slugify(package['title'], separator='_')
-            package['extraction_processors'] = select_processors(package)
-        except ParserError as error:
+        except (ParserError, ScannerError) as error:
             message = clean_error_message(str(error))
             package = {'yaml_error': message}
             logging.debug('Parser error in %s: %s', description_file, message)
 
     with open(datapackage_file, 'w+') as stream:
         stream.write(json.dumps(package, indent=4))
 
-    logging.debug('Created %s', datapackage_file)
+    pipeline_id = os.path.split(source_folder)[-1]
+    logging.info('%s: created datapackage', pipeline_id)
 
     return package
 
 
-def bootstrap_pipeline(source_folder):
-    """Save the pipeline file inside the source folder."""
+def register_processors(processors, pipeline_id):
+    """Register processors according to the datapackage specs."""
 
-    _, source_name = os.path.split(source_folder)
-    package = load_source_datapackage(source_folder)
-    pipeline = {source_name: DEFAULT_PIPELINE}
+    pipeline = {pipeline_id: DEFAULT_PIPELINE}
+    pipeline[pipeline_id]['pipeline'] = processors
+    return pipeline
 
-    if 'yaml_error' not in package:
-        processors = select_processors(package)
-        save_processor_placeholders(processors)
-        save_pipeline(pipeline, source_folder)
 
+def save_pipeline(source_folder, pipeline, pipeline_id):
+    """Save the pipeline-specs.yaml file inside the source folder."""
 
-def save_pipeline(pipeline, source_folder):
-    filepath = os.path.join(source_folder, PIPELINE_SPECS)
+    filepath = os.path.join(source_folder, PIPELINE_FILE)
     with open(filepath, 'w+') as stream:
         stream.write(yaml.dump(pipeline))
-    logging.debug('Created %s', filepath)
+    logging.info('%s: created pipeline specs', pipeline_id)
 
 
 def select_processors(package):
-    """Select the appropriate processors for the pipeline."""
-
-    mode = package['extraction_mode']
-
-    if sum(map(mode.values(), int)) > 1:
-        message = '{}: extraction mode is ambiguous'
-        raise ValueError(message.format(package['name']))
-
-    if mode['download_link']:
-        return ['download_remote_sources']
-    elif mode['web_scraper']:
-        return ['scrape_remote_sources']
-    elif mode['pdf_scraper']:
-        return ['download_pdf_sources', 'scrape_pdf_sources']
-    elif mode['user_interface']:
-        raise NotImplementedError('User interface mode not ready yet.')
+    """Select the appropriate processors for the pipeline with v1."""
+
+    downloader = [
+        {'run': 'validate_source', 'parameters': DATAPACKAGE_FILE},
```

Comment:
Not sure what this is for, but you should try to avoid using external files and encapsulate the details of the datapackage in the pipeline itself.
You can use the metadata and fiscal.model processors to help you do that.